File: server-rails-single/app/models/label.rb
# frozen_string_literal: true

# Label — SHARED GLOBAL catalog in the single-tenant variant.
#
# Labels are NOT user-owned: there is no BelongsToOrganization and no
# Scopes::LabelScope, so every authenticated user sees the same catalog. This
# demonstrates "the user owns everything except shared/global tables."
class Label < Rhino::RhinoModel
  include Discard::Model

  rhino_filters :name
  rhino_sorts :name
  rhino_default_sort "created_at"
  rhino_fields :id, :name, :color, :created_at, :updated_at

  rhino_except_actions :forceDelete

  validates :name, length: { maximum: 255 }, allow_nil: true

  has_and_belongs_to_many :tasks, join_table: :task_labels
end

File: server-laravel-single/app/Models/Label.php
<?php

namespace App\Models;

use Rhino\Models\RhinoModel;

/**
 * Label — SHARED GLOBAL catalog.
 *
 * Labels have no owner (no organization_id, no user_id) and are visible to every
 * authenticated user. There is intentionally NO LabelScope filtering, so the
 * catalog is shared. This demonstrates "the user owns everything except global
 * reference tables."
 */
class Label extends RhinoModel
{
    protected $fillable = [
        'name',
        'color',
    ];

    protected $validationRules = [
        'name' => 'required|string|max:255',
        'color' => 'nullable|string|max:255',
    ];

    public static $allowedFilters = ['name'];
    public static $allowedSorts = ['name'];
    public static $defaultSort = 'created_at';
    public static $allowedFields = ['id', 'name', 'color', 'created_at', 'updated_at'];
    public static $allowedIncludes = [];

    public static array $exceptActions = ['forceDelete'];

    public function tasks(): \Illuminate\Database\Eloquent\Relations\BelongsToMany
    {
        return $this->belongsToMany(Task::class, 'task_labels');
    }
}
